examples/load_balancer/load_balancer.c

```diff
@@ -240,49 +251,83 @@ parse_app_args(int argc, char *argv[], const char *progname) {
-        char mac[32];
-        FILE *cfg;
+parse_backend_json_config(void) {
+        int ret, i;
+        i = 0;
 
-        cfg = fopen(lb->cfg_filename, "r");
-        if (cfg == NULL) {
-                rte_exit(EXIT_FAILURE, "Error openning server \'%s\' config\n", lb->cfg_filename);
+        cJSON *config_json = onvm_config_parse_file(lb->cfg_filename);
+        cJSON *list_size = NULL;
+        cJSON *policy = NULL;
+        cJSON *ip_addr = NULL;
+        cJSON *mac_addr = NULL;
+        cJSON *weight = NULL;
+
+        if (config_json == NULL) {
+                rte_exit(EXIT_FAILURE, "%s file could not be parsed/not found. Assure config file"
+                                       " the directory to the config file is being specified.\n", lb->cfg_filename);
         }
-        ret = fscanf(cfg, "%*s %d", &temp);
-        if (temp <= 0) {
-                rte_exit(EXIT_FAILURE, "Error parsing config, need at least one server configurations\n");
+
+        config_json = config_json -> child;
+
+        list_size = cJSON_GetObjectItem(config_json, "LIST_SIZE");
+        policy = cJSON_GetObjectItem(config_json, "policy");
+        
+        if (list_size == NULL) rte_exit(EXIT_FAILURE, "LIST_SIZE not found/invalid\n");
+        if (policy == NULL) rte_exit(EXIT_FAILURE, "Policy not found/invalid\n");
+
+        lb->server_count = list_size->valueint;
+        lb->policy = strdup(policy->valuestring);
+
+        if (!((!strcmp(lb->policy,"random")) || (!strcmp(lb->policy,"rrobin")) || (!strcmp(lb->policy,"weighted_random")))) {
+                rte_exit(EXIT_FAILURE, "Invalid policy. Check server.conf\n");
         }
-        lb->server_count = temp;
+        
+        
+        lb->weights = (int*)calloc(lb->server_count,sizeof(int));
 
         lb->server = (struct backend_server *)rte_malloc("backend server info",
                                                          sizeof(struct backend_server) * lb->server_count, 0);
         if (lb->server == NULL) {
                 rte_exit(EXIT_FAILURE, "Malloc failed, can't allocate server information\n");
         }
 
-        for (i = 0; i < lb->server_count; i++) {
-                ret = fscanf(cfg, "%s %s", ip, mac);
-                if (ret != 2) {
-                        rte_exit(EXIT_FAILURE, "Invalid backend config structure\n");
-                }
+        config_json = config_json->next;
+
 
-                ret = onvm_pkt_parse_ip(ip, &lb->server[i].d_ip);
+        while (config_json != NULL) {
+                ip_addr = cJSON_GetObjectItem(config_json, "ip");
+                mac_addr =  cJSON_GetObjectItem(config_json, "mac_addr");
+                weight = cJSON_GetObjectItem(config_json, "weight");
+
+                if (ip_addr == NULL) rte_exit(EXIT_FAILURE, "IP not found/invalid\n");
+                if (mac_addr == NULL) rte_exit(EXIT_FAILURE, "MAC address not found/invalid\n");
+                if (weight == NULL) rte_exit(EXIT_FAILURE, "Weight not found/invalid\n");
```

Comment:
as mentioned in #327 we should support config files for policies that don't need a weight. You can fix that here instead of in @xwedea 's version if you want since this will replace his config reading functions.